msu/utils/utils.nut

```diff
@@ -203,6 +203,48 @@
 	    return this.Timers[_id];
 	}
 
+	function benchmark( _functions, _iterations = 100000 )
+	{
+		// _functions can be a function to be benchmarked
+		// or an array of len 2 arrays where idx 0 is an id (can be string) and idx 1 is function to be benchmarked
+		if (typeof _functions == "function")
+			_functions = [["", _functions]];
+
+		local times = array(_functions.len());
+		local timer = ::MSU.Class.Timer("MSU_Benchmark");
+
+		foreach (i, entry in _functions)
+		{
+			local timeStart = timer.silentGet();
+			local f = entry[1];
+
+			for (local j = 0; j < _iterations; j++)
+			{
+				f();
+			}
+
+			local totalTime = timer.silentGet() - timeStart;
+
+			times[i] = {
+				TotalTime = totalTime,
+				TimePerIteration = totalTime / _iterations
+			};
+		}
```

Comment:
I don't think this is how we should be using the timer, just create it and `silentStop` once we're done, that way it handles the subtraction rather than this function.